### DISCUSSION TITLE
Can we generate librclcpp_wasm.a?

### DISCUSSION THREAD

**Discussion by RDaneelOlivav** [RobotWebTools/rclnodejs#933] [General]
Hi,

I was wondering if there is someway to generate all these libraries:

[
    'rclcpp',
    'rcl',
    'rcl_yaml_param_parser',
    'rcutils',
    'rcpputils',
    'rmw',
    'rmw_implementation',
    'tracetools',
    'ament_index_cpp',
    'libstd_msgs__rosidl_typesupport_cpp',
    'libsensor_msgs__rosidl_typesupport_cpp',
    'rosidl_typesupport_cpp',
    'rosidl_typesupport_c',
    'rosidl_runtime_c',
    'liblibstatistics_collector',
    'statistics_msgs__rosidl_typesupport_cpp',
    'rosgraph_msgs__rosidl_typesupport_cpp',
    'rcl_interfaces__rosidl_typesupport_cpp',
    'rcl_interfaces__rosidl_typesupport_c',
    'rcl_interfaces__rosidl_generator_c',
    'rcl_logging_interface',
    'rcl_logging_spdlog',
    'builtin_interfaces__rosidl_generator_c' ]

For being used in Web, at least to be able to link them using WebAssembly?

**Comment by minggangw** [comment]
You can take a look at https://ros2wasm.dev/ @RDaneelOlivav
